src/cloud-api-adaptor/test/e2e/assessment_helpers.go

```diff
@@ -541,47 +541,69 @@ func AssessPodRequestAndLimit(ctx context.Context, client klient.Client, pod *v1
-	var podlist v1.PodList
-	if err := client.Resources(pod.Namespace).List(ctx, &podlist); err != nil {
-		return "Failed to list pod", err
+func AssessPodTestCommands(t *testing.T, ctx context.Context, client klient.Client, pod *v1.Pod, testCommands []TestCommand) error {
+	err := findPod(ctx, client, pod)
+	if err != nil {
+		return err
 	}
 	for _, testCommand := range testCommands {
-		log.Tracef("Running test command: %v", testCommand)
-		var stdout, stderr bytes.Buffer
-		for _, podItem := range podlist.Items {
-			if podItem.ObjectMeta.Name == pod.Name {
-				//adding sleep time to intialize container and ready for Executing commands
-				time.Sleep(5 * time.Second)
-				if err := client.Resources(pod.Namespace).ExecInPod(ctx, pod.Namespace, pod.Name, testCommand.ContainerName, testCommand.Command, &stdout, &stderr); err != nil {
-					if testCommand.TestErrorFn != nil {
-						if !testCommand.TestErrorFn(err) {
-							return err.Error(), fmt.Errorf("command %v running in container %s produced unexpected output on error: %s, stderr: %s", testCommand.Command, testCommand.ContainerName, err.Error(), stderr.String())
-						}
-					} else {
-						return err.Error(), fmt.Errorf("command %v running in container %s produced unexpected output on error: %s, stderr: %s", testCommand.Command, testCommand.ContainerName, err.Error(), stderr.String())
-					}
-				} else if testCommand.TestErrorFn != nil {
-					return "", fmt.Errorf("We expected an error from Pod %s, but it was not found", pod.Name)
-				}
-				if testCommand.TestCommandStderrFn != nil {
-					if !testCommand.TestCommandStderrFn(stderr) {
-						return stderr.String(), fmt.Errorf("Command %v running in container %s produced unexpected output on stderr: %s, stdout: %s", testCommand.Command, testCommand.ContainerName, stderr.String(), stdout.String())
-					} else {
-						return stderr.String(), nil
-					}
-				}
-				if testCommand.TestCommandStdoutFn != nil {
-					if !testCommand.TestCommandStdoutFn(stdout) {
-						return stdout.String(), fmt.Errorf("Command %v running in container %s produced unexpected output on stdout: %s, stderr: %s", testCommand.Command, testCommand.ContainerName, stdout.String(), stderr.String())
-					} else {
-						return stdout.String(), nil
-					}
-				}
+		err := assessPodTestCommand(t, ctx, client, pod, testCommand)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func findPod(ctx context.Context, client klient.Client, pod *v1.Pod) error {
+	var podList v1.PodList
+	if err := client.Resources(pod.Namespace).List(ctx, &podList); err != nil {
+		return fmt.Errorf("Failed to list pod, error : %s", err.Error())
+	}
+	for _, podItem := range podList.Items {
+		if podItem.ObjectMeta.Name == pod.Name {
+			return nil
+		}
+	}
+	return fmt.Errorf("Pod not found with name %s in namespace %s", pod.Name, pod.Namespace)
+}
```

Comment:
nit: Can you move `findPod` above `AssessPodTestCommands` as I think that will help the diff be smaller and make it easier to review